### DISCUSSION TITLE
How do I update a custom field?

### DISCUSSION THREAD

**Discussion by Kealtie** [swellstores/community#477] [Help] [answered]
Hello,

Whilst creating an account through the front-end API. I am trying to add additional content which are stored in custom fields. In this particular case I am trying to add a company_name and a coc_number to the standard customer data.

![image](https://github.com/swellstores/community/assets/7847482/28fba65a-42db-4125-a299-451e1f1dc41c)

I am currently sending the following payload and getting the following response.

Payload:
<img width="300" alt="image" src="https://github.com/swellstores/community/assets/7847482/7bd73cfa-b632-4e84-9f5c-987c8aca3884">

Response:
<img width="512" alt="image" src="https://github.com/swellstores/community/assets/7847482/5fc347da-78c5-408e-8184-006d9406fa0f">

How can I add this custom information to the users account using the front-end API?

**Comment by ericingram** [comment]
The storefront API has a default permission model that is still quite tricky to enable with custom fields. It requires adding a permission config to your public key record. @vstudenichnik-insoft can share an example I think.

The new Swell App framework (currently in alpha) has a new approach where you can simply put `public: true` on a model or field, which makes it accessible through the storefront API. If you want to try it, refer to this announcement: https://github.com/orgs/swellstores/discussions/458

We'll eventually put this flag in the content fields UI in the dashboard. Note you can also modify the field using the developer console and add `public: true` to the field manually.

**Comment by SusannaSwell** [comment]
Speaking about particular case, maybe our standard settings can help:
Enabling business customers (stores the company name) and VAT number. 
It can be found in admin/settings/customers
<img width="1196" alt="Screenshot 2024-01-17 at 16 50 28" src="https://github.com/swellstores/community/assets/79216957/8f40862b-3a22-481f-98b8-dc68c8f559a9">

**Reply by Kealtie** [reply]
Ah, in this particular case we're not trying to get the VAT number, but a COC number (Chamber of Commerce).

"Note you can also modify the field using the developer console and add public: true to the field manually.", How would I do this? I do not have any experience using the developer console within Swell.

And once this is done, how would I have to send my data in the Payload through the front-end API whilst creating an account? Since this information is retrieved whilst the enduser is registering

**Reply by sags95** [reply, marked as answer]
You should be able to open the field for input using this guide here: https://developers.swell.is/guides/opening-content-permissions. After you'll be able to pass the field when creating the account with `swell.account.create`

**Reply by Kealtie** [reply]
Thank you, this did the job
